pkg/kubefed2/federate/federate.go

```diff
@@ -188,24 +186,25 @@ func GetFedResources(hostConfig *rest.Config, qualifiedTypeName, qualifiedName c
 	}, nil
 }
 
-func lookupTypeDetails(config *rest.Config, qualifiedTypeName ctlutil.QualifiedName) (*fedv1a1.FederatedTypeConfig, error) {
-	client, err := genericclient.New(config)
+func lookupTypeDetails(hostConfig *rest.Config, typeName ctlutil.QualifiedName) (*fedv1a1.FederatedTypeConfig, error) {
+	apiResource, err := enable.LookupAPIResource(hostConfig, typeName.Name, "")
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed to get federation client")
+		return nil, err
 	}
+	typeName.Name = typeconfig.GroupQualifiedName(*apiResource)
 
-	typeConfig := &fedv1a1.FederatedTypeConfig{}
-	err = client.Get(context.TODO(), typeConfig, qualifiedTypeName.Namespace, qualifiedTypeName.Name)
+	client, err := genericclient.New(hostConfig)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Error retrieving FederatedTypeConfig %q", qualifiedTypeName)
+		return nil, errors.Wrap(err, "Failed to get federation client")
 	}
 
-	_, err = enable.LookupAPIResource(config, typeConfig.Name, typeConfig.APIVersion)
+	typeConfig := &fedv1a1.FederatedTypeConfig{}
+	err = client.Get(context.TODO(), typeConfig, typeName.Namespace, typeName.Name)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Error retrieving API resource for FederatedTypeConfig %q", qualifiedTypeName)
```

Comment:
Would it make sense to provide special handling for `IsNotFound` and give an indication of how to fix that?

Reply:
Updated, I am not sure if the `recommendation` has the best language.
However I now intend to update this to `Try using --enable-type to also enable the type while federating the resource.` in #663.